app/src/main/java/com/example/vybe/AddEditVibeActivity.java

```diff
@@ -161,9 +163,15 @@ public void onNothingSelected(AdapterView<?> adapterView) {
             //TODO: integrate firestore stuff here i guess
 
             vibeEvent.setDateTime(LocalDateTime.of(selectedDate, selectedTime));
-            vibeEvent.setReason(reasonField.getText().toString());
-
 
+            if (reasonField.getText().toString().trim().split("\\s").length <= reasonMaxWords) {
+                reasonField.setError(null);
+                vibeEvent.setReason(reasonField.getText().toString());
+            } else {
+                // your code here
+                reasonField.setError(String.format("max %d words allowed", reasonMaxWords));
+                return;
```

Comment:
what?

Reply:
return here is used to make sure app doesn't exit activity after clicking the save button and returns process to onCreate